sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/trace/_exporter.py

```diff
@@ -222,21 +251,19 @@ def _convert_span_to_envelope(span: ReadableSpan) -> TelemetryItem:
                         difference = (start_time / 1000000) - enqueued_time
                         total += difference
                 data.measurements["timeSinceEnqueued"] = max(0, total / len(span.links))
-        elif SpanAttributes.HTTP_METHOD in span.attributes:  # HTTP
+        elif HTTP_REQUEST_METHOD in span.attributes or SpanAttributes.HTTP_METHOD in span.attributes:  # HTTP
             path = ""
-            if SpanAttributes.HTTP_USER_AGENT in span.attributes:
+            user_agent = trace_utils._get_user_agent(span.attributes)
+            if user_agent:
                 # TODO: Not exposed in Swagger, need to update def
-                envelope.tags["ai.user.userAgent"] = span.attributes[SpanAttributes.HTTP_USER_AGENT]
-            # http specific logic for ai.location.ip
-            if SpanAttributes.HTTP_CLIENT_IP in span.attributes:
-                envelope.tags[ContextTagKeys.AI_LOCATION_IP] = span.attributes[SpanAttributes.HTTP_CLIENT_IP]
+                envelope.tags["ai.user.userAgent"] = user_agent
```

Comment:
Not new, but we should make this a constant.

Reply:
Oh I remember why I did this explicitly. We generate these constants from swagger but the latest yamls dont have ai.user.userAgent field so we hardcode it until those are updated.

Reply:
Got it